### ISSUE TITLE
Feature Request: Unterstützung für Staffel/Episode Daten

### ISSUE THREAD

**Issue by ChristianKuehnel** [mediathekview/MServer#942]
Das ZDF hat jetzt offenbar die Staffel und Episode bei Serien aus dem Titel entfernt und speichert diese in einem eigenen Feld (Beispiel: https://www.zdf.de/serien/solo-fuer-weiss : "S01 F08").

Es wäre super, wenn MediathekView das auch (z.b. als eigene Spalten) anzeigen würde.

Die ARD entgegen hat diese Daten immer noch im Titel (Beispiel: https://www.ardmediathek.de/serie/der-zuerich-krimi/ "Borcherts Fall (S01/E01)").

**Comment by codingPF** [comment]
Ich glaube hier ist besser aufgdhoben

**Comment by ChristianKuehnel** [comment]
Dann sollte ich mir die Rohdaten mal anschauen und überlegen, ob man daraus irgendwas sinnvolles extrahieren kann.

Woher (welche API/Webseite/...) bezieht denn MediathekView die Inhalte beim ZDF? Ist das ein Webscraper? Oder hat das ZDF eine API dafür?

**Comment by codingPF** [comment]
Dinge ändern sich und es kann nicht schaden (auch bekannte) Themen noch zu durch denken

**Comment by pidoubleyou** [comment]
ja, das ZDF hat die Infos tatsächlich explizit mittlerweile in seiner Datenstruktur: 
![Screenshot from 2023-11-08 21-50-16](https://github.com/mediathekview/MServer/assets/22942659/a6282daa-3f6a-4022-9671-2f281c03f298)

Eine Erweiterung in diese Richtung wäre durchaus mal denkbar.

**Comment by pidoubleyou** [comment]
> Das sieht doch super aus! Kann ich mir diese Daten irgendwo anschauen? Woher beziehen wir diese Daten?

über die ZDF-API. wenn du die Filmseite bei ZDF aufrufst, findest du einen JSON-Request mit folgender Url: https://api.zdf.de/content/documents/zdf/serien/solo-fuer-weiss/solo-fuer-weiss---liebeswut-102.json

Den zugehörigen Deserializer findest du im Code [hier](https://github.com/mediathekview/MServer/blob/develop/src/main/java/de/mediathekview/mserver/crawler/zdf/json/ZdfFilmDetailDeserializer.java).

**Comment by codingPF** [comment]
ich habe mal ein paar Zeilen code rein gebaut...hier der output (random sample) um mal einen Eindruck zu bekommen....NO EPISODE habe ich drin gelassen damit man verstehen kann wie oft das Feld gefüllt ist. Das sieht schon sehr  nett aus aber hat auch eine großen imapact...

```
Blinky Bill#Cranky, der Clown -  (NO EPISODE)
Doku#Vom Freiwilligenjahr zum eigenen Hostel in Afrika - Abenteuer Auswandern in Tansania (NO EPISODE)
In aller Freundschaft#Verpasste Gelegenheiten -  (NO EPISODE)
Bares für Rares#Bares für Rares vom 6. November 2023 - Die Trödel-Show mit Horst Lichter #SE2023#EP256
Löwenzahn#Hühner - Das große Gackern #SE42#EP371
Abendschau - Der Süden#Sendung 09.11.2023 · Das Wichtigste aus Südbayern -  (NO EPISODE)
Death in Paradise#Rückschwung #SE11#EP4
heute#Gedenkfeier zur Reichspogromnacht (NO EPISODE)
ZDFinfo Doku#Murderous History: Der Teufel von Chicago - Rätselhafte Verbrechen #SE2022#EP3
Volle Kanne#Blühendes für den Winter - Tipps von Elmar Mai zu Weihnachtssternen und Amaryllis (NO EPISODE)
SWR1 Leute#Inna Hartwich - Moskau-Korrespondentin - Russland: Darum begehren die Menschen nicht auf -  (NO EPISODE)
Terra X#Welten-Saga - Die Schätze Indiens - Folge 2 der sechsteiligen Dokureihe mit Christopher Clark #SE1#EP2
ZDFinfo Doku#Murderous History: Die Leichenräuber von Edinburgh - Rätselhafte Verbrechen #SE2022#EP2
SWR Aktuell Baden-Württemberg#Sendung 19:30 Uhr vom 6.11.2023 -  (NO EPISODE)
KiKANiNCHEN#Der Krebs -  (NO EPISODE)
Leschs Kosmos#Der Schatz in der Mülltonne - Das Recycling-Versprechen #SE2022#EP7
Insectibles#10. Das große Leuchten -  (NO EPISODE)
heute journal update#Fiedler: "Personal nach wie vor knapp" (NO EPISODE)
Panorama#"The Marvels": Diverser Cast und Actionszenen (NO EPISODE)
Terra X#Überleben – Planet im Wandel - Dokureihe mit Dirk Steffens #SE19#EP5
ZDFinfo Doku#Geheimes Saudi-Arabien - Aufbruch und Unterdrückung  #SE1#EP3
Doku & Reportage#Von der Fabrik zur Kunst -  (NO EPISODE)
```

**Comment by ChristianKuehnel** [comment]
Danke für die Links!

Ich versuche mal zusammenzufassen, was zu tun wäre:
* Crawler:
  * Vom ZDF bekommen wir bei einigen Serien/Folgen die Daten geliefert, bei anderen nicht.
  * Bei der ARD könnte man dieselbe Information mit einer Heuristik `(S01/E02)` aus dem Titel raten. Wird manchmal funktionieren und manchmal auch nicht.
  * Weitere Sender müsste man sich einzeln anschauen.
* Datenmodell:
  * Man müsste das gesamte Datenmodell an diversen Stellen um die entsprechenden Attribute erweitern.
  * Nachdem das eine größere Änderung ist müsste man das irgendwie rückwärtskompatibel gestalten, insbesondere damit ältere SW-Versionen sich nicht an den neuen Daten verschlucken.
* Frontends:
  * Man müsste das Java und das Webfrontend erweitern, damit dort die zusätzlichen Daten dort auch angezeigt und gefiltert/sortiert werden können.

Das klingt tatsächlich nach einer größeren Aktion. Seht ihr da einen Mehrwert, der den Aufwand rechtfertigt?
